examples/mnist.py

```diff
@@ -1,10 +1,20 @@
 from __future__ import print_function
 
-import cPickle as pickle
 import gzip
 import itertools
-import urllib
+import pickle
 import os
+import sys
+
+PY2 = sys.version_info[0] == 2
+
+if PY2:
+    from urllib import urlretrieve
+    pickle_load = lambda f, encoding: pickle.load(f)
+else:
+    from urllib.request import urlretrieve
+    pickle_load = lambda f, encoding: pickle.load(f, encoding=encoding)
+
```

Comment:
Not a case of disk caching here.  Try for yourself, run the lines in reverse order, you'll get consistent results.